master/buildbot/process/remotecommand.py

```diff
@@ -212,7 +242,20 @@ def remote_update(self, updates):
             # log.msg("update[%d]:" % num)
             try:
                 if self.active and not self.ignore_updates:
-                    self.remoteUpdate(update)
+                    for key, value in update.items():
```

Comment:
sounds like the exact same code as earlier. needs refactor

Reply:
It is intended.

A subsequent commit will move splitting the lines on msgpack to worker side, so earlier code, which now seems to be repeating, will be removed. Only in PB protocol splitting the lines will be on master side.